app/workers/databases.php

```diff
@@ -268,7 +280,13 @@ protected function createIndex(Document $database, Document $collection, Documen
             $dbForProject->updateDocument('indexes', $index->getId(), $index->setAttribute('status', 'available'));
         } catch (\Throwable $th) {
             Console::error($th->getMessage());
-            $dbForProject->updateDocument('indexes', $index->getId(), $index->setAttribute('status', 'failed'));
+            $dbForProject->updateDocument(
```

Comment:
Some errors might expose sensitive server information. I suggest to only catch errors of a specific type, and that we have thrown ourselves. Avoiding native PDO errors that might expose information that we should keep private.

This might require us to add some base Database Exception that we could use to distinguish between our errors and native errors. This are the database exceptions we support today: https://github.com/utopia-php/database/tree/main/src/Database/Exception

Reply:
https://github.com/utopia-php/database/pull/218 has a base exception class implemented.